packages/ra-ui-materialui/src/input/DateTimeInput.spec.tsx

```diff
@@ -223,7 +223,7 @@ describe('<DateTimeInput />', () => {
             const onSubmit = jest.fn();
             render(
                 <AdminContext dataProvider={testDataProvider()}>
-                    <SimpleForm onSubmit={onSubmit}>
+                    <SimpleForm mode="onBlur" onSubmit={onSubmit}>
```

Comment:
Now this worries me a bit. Why did you need to change this test?

Reply:
Well, it seems that the save button wasn't being enabled, which, should be right, I think it was wrong before

Reply:
It's really weird that the test passed before this PR then.

But you convinced me your fix is right. Let's keep it.

Reply:
Actually, my explanation of why this test should be right wasn't precise, is not relevant that the input is required, but that is being cleared and back to not being dirty, which is what matters to the SaveButton disable state